include/unifex/detach_on_cancel.hpp

```diff
@@ -0,0 +1,204 @@
+  auto tryGetOp() noexcept {
+    // if cancellation won, parentOp_ would be null at this point
+    // see operation_state::cancel_callback()
+    auto* op = state_->parentOp_.exchange(nullptr, std::memory_order_acq_rel);
+    if (op != nullptr) {
+      op->callback_.destruct();
+    } else {
+      delete state_;
+    }
+    return op;
+  }
+
+  template <typename... Values>
+  void set_value(Values&&... values) noexcept {
+    if (auto op = tryGetOp()) {
+      UNIFEX_TRY {
+        unifex::set_value(std::move(op->receiver_), (Values &&) values...);
+      }
+      UNIFEX_CATCH(...) {
+        unifex::set_error(std::move(op->receiver_), std::current_exception());
+      }
+    }
+  }
+
+  template <typename Error>
+  void set_error(Error&& error) noexcept {
+    if (auto op = tryGetOp()) {
+      unifex::set_error(std::move(op->receiver_), (Error &&) error);
+    }
+  }
+
+  void set_done() noexcept {
+    if (auto op = tryGetOp()) {
+      unifex::set_done(std::move(op->receiver_));
+    }
+  }
+
+  friend inplace_stop_token
+  tag_invoke(tag_t<get_stop_token>, const _receiver& r) noexcept {
+    return r.state_->stopSource_.get_token();
+  }
+  detached_state* state_;
+};
+
+template <typename UpstreamSender, typename DownstreamReceiver>
+struct operation_state<UpstreamSender, DownstreamReceiver>::detached_state
+    final {
+  detached_state(
+      typename operation_state<UpstreamSender, DownstreamReceiver>::type* op,
+      UpstreamSender&&
+          s) noexcept(is_nothrow_connectable_v<UpstreamSender, _receiver>)
+    : parentOp_(op)
+    , childOp_(connect(std::move(s), _receiver{this})){};
+
+  std::atomic<
+      typename operation_state<UpstreamSender, DownstreamReceiver>::type*>
+      parentOp_;
+  inplace_stop_source stopSource_;
+  connect_result_t<UpstreamSender, _receiver> childOp_;
+};
+
+template <typename Sender>
+struct _sender {
+  struct type;
+};
+
+template <typename Sender>
+using sender = typename _sender<Sender>::type;
+
+template <typename Sender>
+struct _sender<Sender>::type {
+  UNIFEX_NO_UNIQUE_ADDRESS Sender upstream_sender_;
+
+  template <
+      template <typename...>
+      class Variant,
+      template <typename...>
+      class Tuple>
+  using value_types =
+      typename sender_traits<Sender>::template value_types<Variant, Tuple>;
+
+  template <template <typename...> class Variant>
+  using error_types = typename concat_type_lists_unique_t<
+      sender_error_types_t<
+          typename sender_traits<Sender>::template error_types<Variant>,
+          type_list>,
+      type_list<std::exception_ptr>>::template apply<Variant>;
+
+  static constexpr bool sends_done = true;
+
+  friend constexpr auto
+  tag_invoke(tag_t<blocking>, const type& sender) noexcept {
+    auto block_value = blocking(sender.upstream_sender_);
+    if (block_value == _block::_enum::never) {
+      return _block::_enum::maybe;
+    }
+    return block_value;
```

Comment:
This isn't _quite_ right, because `blocking(sender.upstream_sender_)` might return an instance of `std::integral_constant` rather than an instance of `unifex::blocking`.  I think you need something like this:
```suggestion
    if constexpr (same_as<blocking_kind, decltype(blocking(sender.upstream_sender_))>) {
      // the sender returns a runtime-determined blocking_kind
      blocking_kind blockValue = blocking(sender.upstream_sender_);
      if (blockValue == blocking_kind::never) {
        blockValue = blocking_kind::maybe;
      }
      return blockValue;
    }
    else if constexpr (blocking_kind::never == cblocking<Sender>()) {
      // the sender always returns never
      return blocking_kind::maybe;
    }
    else {
      return cblocking<Sender>();
    }      
```
You may need @ericniebler's eyes on this one.